PayItForward.Data.Services/UsersService.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace PayItForward.Services.Data
+{
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using PayItForward.Data;
+    using PayItForward.Services.Data.Abstraction;
+    using Dbmodel = PayItForward.Data.Models;
+
+    // TO DO :
+    // Bisness logic implementation later when Constructors are created
+
+    public class UsersService : IUsersService
+    {
+        private readonly UsersRepository<Dbmodel.User, string> usersRepo;
+
+        public UsersService()
+        {
+            this.usersRepo = new UsersRepository<Dbmodel.User, string>(new PayItForwardDbContext());
+        }
+
+        public IQueryable<Dbmodel.User> GetAll()
+        {
+            return this.usersRepo.GetAll().OrderBy(u => u.Id);
+        }
+
+        public int Count()
+        {
+            return this.usersRepo.GetAll().Count();
+        }
+
+        public async Task<Dbmodel.User> GetByIdAsync(string id)
+        {
+            return await this.usersRepo.GetByIdAsync(id);
+        }
+
+        public Dbmodel.User GetById(string id)
+        {
+            return this.usersRepo.GetById(id);
+        }
+
+        public Dbmodel.User GetByUserName(string userName)
+        {
+            return this.usersRepo.GetAll().FirstOrDefault(u => u.UserName == userName);
+        }
+
+        public Task SaveAsync()
```

Comment:
The service should not have save changes method, but instead Adduser method for example, as this is business logic but not CRUD operations